waku/waku_api/handlers.nim

```diff
@@ -0,0 +1,50 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  chronos,
+  chronicles,
+  std/[options, sequtils],
+  stew/results
+import
+  ../waku_discv5,
+  ../waku_relay,
+  ../waku_core,
+  ./message_cache
+
+### Discovery
+
+type DiscoveryHandler* = proc(): Future[Result[Option[RemotePeerInfo], string]] {.async, closure.}
+
+proc defaultDiscoveryHandler*(discv5: WakuDiscoveryV5, cap: Capabilities): DiscoveryHandler =
+  proc(): Future[Result[Option[RemotePeerInfo], string]] {.async, closure.} =
+    #Discv5 is already filtering peers by shards no need to pass a predicate.
+    let findPeers = discv5.findRandomPeers()
+    
+    if not await findPeers.withTimeout(60.seconds):
```

Comment:
Thank you! I think it is ok for now. Maybe later we can make it configurable. At least I think it is really on the application or network what is the acceptable timeframe to get connected.

Reply:
Searching for a specific peer in a decentralized network can take long time to maybe not find anything at all.